web-admin/src/features/organizations/settings/UpdateOrgItem.svelte

```diff
@@ -0,0 +1,145 @@
+  import type { AxiosError } from "axios";
+  import { defaults, superForm } from "sveltekit-superforms";
+  import { yup } from "sveltekit-superforms/adapters";
+  import { object, string } from "yup";
+  import Input from "@rilldata/web-common/components/forms/Input.svelte";
+
+  export let organization: string;
+
+  const initialValues: {
+    name: string;
+    description: string;
+  } = {
+    name: "",
+    description: "",
+  };
+  const schema = yup(
+    object({
+      name: string().required(),
+      description: string(),
+    }),
+  );
+
+  const updateOrgMutation = createAdminServiceUpdateOrganization();
+
+  const { form, errors, enhance, submit } = superForm(
+    defaults(initialValues, schema),
+    {
+      SPA: true,
+      validators: schema,
+      async onUpdate({ form }) {
+        if (!form.valid) return;
+        const values = form.data;
+
+        const newOrg = sanitizeOrgName(values.name);
+
+        try {
+          await $updateOrgMutation.mutateAsync({
+            name: organization,
+            data: {
+              newDisplayName: values.name,
+              newName: newOrg,
+              description: values.description,
+            },
+          });
+        } catch (err) {
+          const parsedErr = parseUpdateOrgError(err);
+          if (parsedErr.duplicateOrg) {
+            form.errors.name = [`The name ${newOrg} is already taken`];
+          }
+          return;
+        }
+
+        if (organization !== newOrg) {
+          queryClient.removeQueries(
+            getAdminServiceGetOrganizationQueryKey(organization),
+          );
+          setTimeout(() => goto(`/${newOrg}/-/settings`));
+        } else {
+          void queryClient.refetchQueries(
+            getAdminServiceGetOrganizationQueryKey(organization),
+          );
+        }
+        eventBus.emit("notification", {
+          message: "Updated organization",
+        });
+      },
+      resetForm: false,
+    },
+  );
+
+  $: orgResp = createAdminServiceGetOrganization(organization);
+  $: if ($orgResp.data?.organization) {
+    $form.name =
+      $orgResp.data.organization.displayName || $orgResp.data.organization.name;
+    $form.description = $orgResp.data.organization.description;
+  }
+
+  $: changed =
+    $orgResp.data?.organization?.name !== $form.name ||
+    $orgResp.data?.organization?.description !== $form.description;
+
+  $: error = parseUpdateOrgError(
+    $updateOrgMutation.error as unknown as AxiosError<RpcStatus>,
+  );
+</script>
+
+<SettingsItemContainer title="Organization">
+  <form
+    slot="description"
+    id="org-update-form"
+    on:submit|preventDefault={submit}
+    class="update-org-form"
+    use:enhance
+  >
+    <Input
+      bind:value={$form.name}
+      errors={$errors?.name}
+      id="name"
+      label="Name"
+      description={`Your org URL will be https://ui.rilldata.com/${sanitizeOrgName($form.name)}, to comply with our naming rules.`}
```

Comment:
Nit: it might save some confusion in the dev/test/stage environments if you used the `RILL_ADMIN_FRONTEND_URL`